test/index.js

```diff
@@ -464,6 +464,64 @@ describe('def', function() {
                    'as its first argument; received 0.5'));
   });
 
+  it('supports enumerated types', function() {
+    //  TimeUnit :: Type
+    var TimeUnit = $.EnumType(['milliseconds', 'seconds', 'minutes', 'hours']);
+
+    var def = $.create($.env.concat([TimeUnit, $.ValidDate, $.ValidNumber]));
+
+    //  convertTo :: TimeUnit -> ValidDate -> ValidNumber
+    var convertTo =
+    def('convertTo',
+        {},
+        [TimeUnit, $.ValidDate, $.ValidNumber],
+        function recur(unit, date) {
+          switch (unit) {
+            case 'milliseconds': return date.valueOf();
+            case 'seconds':      return recur('milliseconds', date) / 1000;
+            case 'minutes':      return recur('seconds', date) / 60;
+            case 'hours':        return recur('minutes', date) / 60;
+          }
+        });
+
+    throws(function() { convertTo('days', new Date(0)); },
+           errorEq(TypeError,
+                   '‘convertTo’ expected a value of type ' +
+                   '("milliseconds" | "seconds" | "minutes" | "hours") as ' +
+                   'its first argument; received "days"'));
```

Comment:
Not at all, a lot of these were silly mistakes I should have caught myself. I'm just going to blame tiredness and thus absolve myself of all culpability. ;)